experiment/process-e2e-suite-log/workGroup.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"sync"
+)
+
+// ProcessFileFunction is an alias for a function that will process the file
+// given as an argument.
+type ProcessFileFunction func(fileName string)
+
+// quantity of workers to use in processing
+var workersQuantity int
```

Comment:
Please avoid global variables. It doesn't look like `workersQuantity` is used outside `SetupWorkGroup` and you can pass `fileNameWaitingGroup` and `fileNameChannel` into `worker()`